src/logging/Kconfig

```diff
@@ -33,4 +33,11 @@ config LOG_BACKEND_SOF_PROBE_OUTPUT
         help
                 Set the logging format for SOF probe output.
 
+config LOG_BACKEND_SOF_PROBE_OUTPUT_AUTO_ENABLE
+       bool "Automatically enabe probes logging when probe dma is started"
+       depends on LOG_BACKEND_SOF_PROBE_OUTPUT
+       default y
```

Comment:
This cant be default y, we can have this feature as optional if you want but in normal flow we need to add probe point to enable logging.

Reply:
Ok, I removed the "default y". Can you explain why we can not have it? Does Windows driver start to probe extract DMA automatically, or something?

Reply:
Because we will try to add probe point to start logging and it will result in error because the same probe point was already added by auto_enable feature.

Please make sure kernel is not assuming it will auto start and send proper request to start logging.

Reply:
At the moment kernel is not assuming anything. Its totally agnostic about the probe points and just sends the magic numbers fed to the debugfs file in an to the firmware. At the moment its up to the user to come up with the correct magic numbers to enable the probes it wants, and the number are different in IPC3 and IPC4 modes, that is why I wanted hide the complexity from the user in the simple logging phase.

Reply:
Anyway, there is still three things the user must do for the logs to be coming through probes interface.
1. Turn on probes logging back end in fw build
2. Turn on probes auto logging in fw build
3. Start probes DMA

If someone just likes to user probes for something else and even if we would have the logging back end and the auto logging on by default, it would hardly matter that he gets the logs through probes too. And in any case its easy enough leave the probes logging options out.

But then again I am new here and may not understand everything. Just make a concrete proposal and I change this. So were you suggesting a special debugfs file to turn logs on. @lgirdwood , @ujfalusi , @ranj063  does this sound acceptable? Should I add that to linux driver and drop this PR?